ksql-engine/src/test/resources/query-validation-tests/project-filter.json

```diff
@@ -94,6 +94,27 @@
           "timestamp": 0
         }
       ]
+    },
+    {
+      "name": "Json Multi Dimensional Array",
+      "statements": [
+        "CREATE STREAM array_array (ID BIGINT, ARRAY_COL ARRAY<ARRAY<VARCHAR>>) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM S3 as SELECT ID, ARRAY_COL[0][1] AS array_item FROM array_array;"
```

Comment:
What's the error if the `SELECT` is on `ARRAY_COL[0][3]`? Do we log it? Do we just log an `ArrayOutOfBoundsException`? IF the user is doing this on a CLI, what do they see? 

This is lacking negative test cases or any discussion of the UX for the failing path.

Reply:
The out of bound error is handled the same way we handle other errors while computing the expressions. We log the error and return `null`.

Reply:
What will the user see on the CLI? Is it a useful message? Is there a test for the negative case that exemplifies this message?

Reply:
No, user won't see any messages since this is per row. The error message along with the exception details will be in the log.